packages/app/src/cli/services/dev/urls.ts

```diff
@@ -61,7 +61,10 @@ export async function generateFrontendURL(options: FrontendURLOptions): Promise<
   }
 
   if (environment.spin.isSpin() && !options.tunnelUrl) {
-    frontendUrl = `https://cli.${await environment.spin.fqdn()}`
+    frontendPort = environment.spin.appPort() ?? frontendPort
+    frontendUrl = environment.spin.appHost()
+      ? `https://${environment.spin.appHost()}`
+      : `https://cli.${await environment.spin.fqdn()}`
     return {frontendUrl, frontendPort, usingLocalhost}
   }
```

Comment:
Have we considered providing an interface that can be used not only by the Spin setup but also by 3P developers?

```
--tunnel-url  <-- We mark it as deprecated
--url ${SPIN_APP_HOST} --port ${SPIN_APP_PORT}
```

Reply:
But in case we are inside a `spin environment` we want to apply this rule automatically, right?

Reply:
Applying things automatically is what we've been doing so your work aligns with it. The problem that I see with continuing with that approach is that 1P would benefit from an interface for customizing these things, but they can't because we are coupling some internals to our Spin use case. In other words, if we think more broadly, we'll end up with a CLI that's more flexible and can work in more scenarios.

Reply:
Hmmm, what scenarios would this unlock, @pepicrft? I'm with you in theory, but I need help understanding the impact/effort here. cc: @stephszeto